### PR TITLE
Introduce testable examples

### LINKED ISSUE #163: Update examples to be compilable.
It was decided upon to create an examples directory, located in ``source`` (required for ``literalinclude``). A CI test will be created to build the examples included there against WPILib. The website can still show snippets of the documents using the [literalinclude](https://www.sphinx-doc.org/en/1.5/markup/code.html#directive-literalinclude) directive. 

Since Sphinx Code Tabs do not support text, we will use a text tab ``tab`` and use the literalinclude with an absolute path, with line numbers.

Examples should be located in a ``/examples/java`` or a ``/examples/cpp`` directory. They should be as compact as possible, and should avoid referencing other classes.

```
.. literalinclude:: /examples/example.cpp
   :language: cpp
   :lines: 33-39
   :linenos:
   :lineno-start: 33
```

in which we get this

image

### PR DESCRIPTION
**Pull request by Daltz333**
- Remove WPILib.h references
- Introduce examples as literal includes
- Update style guide
- Small formatting fixes

Closes #163

### CODE REVIEW

File: source/docs/software/vision-processing/introduction/using-the-cameraserver-on-the-roborio.rst [hunk 1]
```diff
@@ -10,108 +10,44 @@ The following program starts automatic capture of a USB camera like the Microsof
...
-          {
-            CameraServer::GetInstance()->StartAutomaticCapture();
-          }
-        };
-        START_ROBOT_CLASS(Robot)
+        .. literalinclude:: examples/using-the-cameraserver-on-the-roborio-1/cpp/Robot.cpp
+           :language: cpp
+           :lines: 11-15
+           :linenos:
+           :lineno-start: 21
```
Comment (AustinShalit) [inline source/docs/software/vision-processing/introduction/using-the-cameraserver-on-the-roborio.rst]: Is this correct?

**Comment by Daltz333** [comment]
As per conversation. The header will be moved into the .cpp file.

File: source/docs/software/vision-processing/introduction/using-the-cameraserver-on-the-roborio.rst (hunk 1 above)
Comment (Daltz333) [reply source/docs/software/vision-processing/introduction/using-the-cameraserver-on-the-roborio.rst]: that lineno-start isn't 😅

**Comment by AustinShalit** [comment]
I think this can be closed because we are doing remote includes from allwpilib?

**Comment by Daltz333** [comment]
A separate PR will be created adding styleguide information

**Pull request closed without merging**